### PR TITLE
interfaces/serial-port: add ttyGS to serial port allow list

### PR DESCRIPTION
**Pull request by tsunghanliu**
Add support for USB gadget serial devices.
    
ttyGSX: drivers/usb/gadget/legacy/g_serial.c
Document: https://www.kernel.org/doc/Documentation/usb/gadget_serial.txt

I also changed the variable IDs, so it would be easier to add new serial devices.

### CODE REVIEW

File: interfaces/builtin/serial_port.go [hunk 1]
```diff
@@ -75,7 +75,8 @@ func (iface *serialPortInterface) String() string {
 //  - ttySCX (NXP SC16IS7xx serial devices)
 //  - ttyMSMX (Qualcomm msm7x serial devices)
 //  - ttyHSX (Qualcomm GENI based QTI serial cores)
-var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty(mxc|USB|ACM|AMA|XRUSB|S|O|SC|MSM|HS)[0-9]+$")
+//  - ttyGS (USB gadget serial devices)
```
Comment (mardy) [inline interfaces/builtin/serial_port.go:78]: nitpick: for consistency with the above, please write `ttyGSX`

File: interfaces/builtin/serial_port_test.go [hunk 2]
```diff
@@ -92,6 +94,8 @@ type SerialPortInterfaceSuite struct {
 	badPathSlot12Info    *snap.SlotInfo
 	badPathSlot13        *interfaces.ConnectedSlot
 	badPathSlot13Info    *snap.SlotInfo
+	badPathSlot100        *interfaces.ConnectedSlot
+	badPathSlot100Info    *snap.SlotInfo
```
Comment (mardy) [inline interfaces/builtin/serial_port_test.go:98]: the indentation is wrong here

**Review by mardy — COMMENTED** [review]
Just some nitpicks (but the indentation one must be fixed, or CI will fail), otherwise looks fine to me. Thanks!

File: interfaces/builtin/serial_port.go (hunk 1 above)
Comment (tsunghanliu) [reply interfaces/builtin/serial_port.go]: thanks & updated

File: interfaces/builtin/serial_port_test.go (hunk 2 above)
Comment (tsunghanliu) [reply interfaces/builtin/serial_port_test.go]: thanks & updated

**Review by mardy — APPROVED** [review]
LGTM, thanks!

**Review by mvo5 — APPROVED** [review]
Thank you

**Comment by tsunghanliu** [comment]
Hi @mvo5,
May I ask for adding this to 2.56,x ? Thank you.

**Pull request merged**